linera-core/src/worker.rs

```diff
@@ -1369,6 +1420,9 @@ where
             info.requested_hashed_certificate_value =
                 Some(self.storage.read_hashed_certificate_value(hash).await?);
         }
+        if let Some(blob_id) = query.request_blob {
+            info.requested_blob = Some(self.storage.read_blob(blob_id).await?);
```

Comment:
Are we writing the blob anywhere? Should probably happen in `process_confirmed_block`.

Reply:
Yes, coming up